tests/test_unsupported_python_version.py

```diff
@@ -0,0 +1,19 @@
+import sys
+from io import StringIO
+from unittest import TestCase
+
+from pytest import mark
+
+
+class UnsupportedPythonVersionTestCase(TestCase):
+
+    @mark.skipif(sys.version_info >= (3, 6), reason='only meant for unsupported Python versions')
+    def test_unsupported_python_version(self):
```

Comment:
It can seem pointless, but it will allow our test coverage to eventually reach 100%, and as long as we keep the Travis CI job in Python 3.5 even after dropping 3.6 and later versions, it will prevent us from breaking this message, e.g. by switching it to an f-string.

Reply:
That's a very comprehensive approach, I'm impressed -  I've never seen a project testing against Python versions it doesn't support :) 

My main worries were about the CI time and maintenance time. 

CI time doesn't seem to be large, but that's still a minute per job. 

Regarding maintenance: installing of old Pythons may stop working, pip may stop working in older Pythons, having 2 more envs increases chances for random failure of a build (e.g. because of a network issue in one of the jobs). Also, we now use https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires in setup.py, so that the error message which we could break may become less and less relevant over time. 

Because of that, +0.5 from me for **not** testing against Python 2.7 and 3.5.